drivers/gpio/gpio_tle9104.c

```diff
@@ -0,0 +1,533 @@
+/*
+ * Copyright (c) 2023 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT infineon_tle9104
+
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(gpio_tle9104, CONFIG_GPIO_LOG_LEVEL);
+
+#define TLE9104_RESET_DURATION_TIME_IN_US                    10
```

Comment:
It would be best to start this file with a comment documenting where the various values and definitions were sourced from. This applies to constants like these as well as the register definitions. A link to a datasheet or other source would be acceptable.

Reply:
done